elk/training/sweep.py

```diff
@@ -57,10 +72,29 @@ def execute(self):
                 # Allow for multiple datasets to be specified in a single string with
                 # plus signs. This means we can pool datasets together inside of a
                 # single sweep.
-                datasets = [ds.strip() for ds in dataset_str.split("+")]
+                train_datasets = [ds.strip() for ds in dataset_str.split("+")]
 
                 run = deepcopy(self.run_template)
                 run.data.model = model_str
-                run.data.prompts.datasets = datasets
+                run.data.prompts.datasets = train_datasets
                 run.out_dir = out_dir
                 run.execute()
+
+                if len(eval_datasets) > 1:
+                    print(colorize("== Transfer eval ==", "green"))
+
+                # Now evaluate the reporter on the other datasets
+                for eval_dataset in eval_datasets:
+                    # We already evaluated on this one during training
+                    if eval_dataset in train_datasets:
+                        continue
+
+                    eval = Eval(
+                        data=Extract(
+                            model=model_str,
+                            prompts=PromptConfig(datasets=[eval_dataset]),
```

Comment:
I think the Extract config here should be initialized with a deepcopy of the base Extract config, because currently `max_examples` and other extraction args don't affect evals. Also, this can help the cache performance of extraction.

Reply:
Fixed